corehq/apps/case_importer/tracking/views.py

```diff
@@ -32,8 +37,28 @@ def to_user_json(case_upload):
             'user': get_display_name_for_user_id(
                 domain, case_upload.couch_user_id, default=''),
             'case_type': case_upload.case_type,
+            'upload_file_name': case_upload.upload_file_name,
+            'upload_file_length': case_upload.upload_file_length,
+            'upload_file_download_allowed': user_may_view_file_upload(
+                domain, request.couch_user, case_upload)
         }
 
     case_uploads = [to_user_json(case_upload)
                     for case_upload in get_case_uploads(domain, limit)]
     return json_response(case_uploads)
+
+
+@require_can_edit_data
+def case_upload_file(request, domain, upload_id):
+    try:
+        case_upload = CaseUploadRecord.objects.get(upload_id=upload_id, domain=domain)
+    except CaseUploadRecord.DoesNotExist:
+        return HttpResponseNotFound()
+
+    if not user_may_view_file_upload(domain, request.couch_user, case_upload):
+        return HttpResponseForbidden()
+
+    response = StreamingHttpResponse(open(case_upload.get_tempfile_ref_for_upload_ref(), 'rb'))
```

Comment:
I still think it would be better to let nginx handle this than to tie up resources on a web worker.